scripts/needs_testrun.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+from argparse import ArgumentParser
+import fnmatch
+import json
+from pathlib import Path
+from subprocess import check_output
+import sys
+import typing as t
+
+
+def get_changed_files() -> t.List[str]:
```

Comment:
Doing this with a subprocess `git diff` call might be slightly less "correct" but it would get around the issue of needing a Github token in CircleCI.

Reply:
The `git diff` would work for single-commit PRs. In general we would need to know the target branch to compute the full diff, but that information is available only at the PR level 🙁 .

Reply:
I've added calls to the REST API to get accurate results and used the `git diff` as a fallback

Reply:
nice, then we can at least see what kind of API limits we are hitting.